service/dap/server.go

```diff
@@ -735,10 +736,29 @@ func (s *Server) setClientCapabilities(args dap.InitializeRequestArguments) {
 	s.clientCapabilities.supportsVariableType = args.SupportsVariableType
 }
 
-// Default output file pathname for the compiled binary in debug or test modes,
-// relative to the current working directory of the server.
+// Default output file pathname for the compiled binary in debug or test modes
+// when temporary debug binary creation fails.
+// This is relative to the current working directory of the server.
 const defaultDebugBinary string = "./__debug_bin"
 
+func (s *Server) tempDebugBinary() string {
+	binaryPattern := "__debug_bin"
+	if runtime.GOOS == "windows" {
+		binaryPattern = "__debug_bin*.exe"
+	}
+	f, err := ioutil.TempFile("", binaryPattern)
+	if err != nil {
+		s.log.Errorf("failed to create a temporary binary (%v), falling back to %q", err, defaultDebugBinary)
+		return defaultDebugBinary
+	}
+	name := f.Name()
+	if err := f.Close(); err != nil {
+		s.log.Errorf("failed to create a temporary binary (%v), falling back to %q", err, defaultDebugBinary)
+		return defaultDebugBinary
```

Comment:
It's confusing that you use exe with the temp file name, but not default one here. How about using cleanExeName at the end only.

Reply:
That's ioutil.TempFile's search pattern to avoid conflicts. https://pkg.go.dev/io/ioutil#TempFile

Reply:
Then you should use cleanExeName here on the default binary. And I wonder if you should use it on the user-specified value.

Reply:
Done.
We end up here only if there is no user-specified value.